src/spotify.rs

```diff
@@ -0,0 +1,58 @@
+use rspotify::{
+    model::{Country, Market, SearchType, TrackId},
+    prelude::*,
+    ClientCredsSpotify,
+};
+
+use crate::track::Track;
+
+#[derive(Clone)]
+pub struct SpotifyAPIClient {
+    inner: ClientCredsSpotify,
+}
+
+impl SpotifyAPIClient {
+    pub async fn new(spotify_client_id: &str, spotify_client_secret: &str) -> anyhow::Result<Self> {
+        let spotify_api_client = rspotify::ClientCredsSpotify::with_config(
+            rspotify::Credentials::new(spotify_client_id, spotify_client_secret),
+            rspotify::Config {
+                token_refreshing: true,
+                ..rspotify::Config::default()
+            },
+        );
+        if let Err(err) = spotify_api_client.request_token().await {
+            Err(anyhow::anyhow!(err))
+        } else {
+            Ok(Self {
+                inner: spotify_api_client,
+            })
+        }
+    }
+
+    pub async fn get_by_id(&self, track_id: &str) -> anyhow::Result<Track> {
+        let track_id = TrackId::from_id(track_id)?;
+        let track = &self.inner.track(track_id, None).await?;
+        Ok(Track::from(track))
+    }
+
+    pub async fn search(&self, track_query: &str) -> anyhow::Result<Vec<Track>> {
+        tracing::info!("searching for \"{}\"...", track_query);
+        let result = &self
+            .inner
+            .search(
+                track_query,
+                SearchType::Track,
+                Some(Market::Country(Country::Ukraine)),
+                None,
+                Some(10),
+                None,
+            )
+            .await?;
+
+        if let rspotify::model::SearchResult::Tracks(tracks) = result {
```

Comment:
`return` is really there for early returns, for bounded function exit points it's better to work with the compiler...

```rust
match result {
	rspotify::model::SearchResult::Tracks(x) => x.items.iter().map(Track::from).collect(),
	_ => vec![]
}
 ```